tests/components/enphase_envoy/test_config_flow.py

```diff
@@ -397,6 +399,186 @@ async def test_zeroconf_host_already_exists(
     assert config_entry.title == "Envoy 1234"
 
 
+async def test_zero_conf_while_form(
+    hass: HomeAssistant, config_entry, setup_enphase_envoy
+) -> None:
+    """Test zeroconf while form is active."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("1.1.1.1"),
+            ip_addresses=[ip_address("1.1.1.1")],
+            hostname="mock_hostname",
+            name="mock_name",
+            port=None,
+            properties={"serialnum": "1234", "protovers": "7.0.1"},
+            type="mock_type",
+        ),
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.ABORT
```

Comment:
https://github.com/home-assistant/core/pull/114405/commits/28c6bf18b39444a44fa4d6163326a0cb834ed579#r1566299667

Reply:
added assert for reason